picard/discid/eaclog.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+#
+# MIT License
+#
+# Copyright(c) 2018 Konstantin Mochalov
+# Copyright(c) 2022 Philipp Wolfer
+#
+# Original code from https://gist.github.com/kolen/765526
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files(the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import re
+
+
+class NotSupportedTOCError(Exception):
+    pass
+
+
+def filter_toc_entries(lines):
+    """
+    Take iterator of lines, return iterator of toc entries
+    """
+
+    # Search the TOC table header
+    for line in lines:
+        # to allow internationalized EAC output where column headings
+        # may differ
+        if re.match(r""" \s*
+                   .+\s+ \| (?#track)
+                \s+.+\s+ \| (?#start)
+                \s+.+\s+ \| (?#length)
+                \s+.+\s+ \| (?#start sec)
+                \s+.+\s*$   (?#end sec)
+                """, line, re.X):
+            # Skip over the table header separator
+            next(lines)
+            break
+
+    for line in lines:
+        m = re.match(r"""
+            ^\s*
+            (?P<num>\d+)
+            \s*\|\s*
+            (?P<start_time>[0-9:.]+)
+            \s*\|\s*
+            (?P<length_time>[0-9:.]+)
+            \s*\|\s*
+            (?P<start_sector>\d+)
+            \s*\|\s*
+            (?P<end_sector>\d+)
+            \s*$
+            """, line, re.X)
```

Comment:
Since that's in a loop, it would be better to compile the regex before entering the loop.